src/main/java/com/powsybl/openloadflow/ac/outerloop/tap/TransformerRatioManager.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.ac.outerloop.tap;
+
+import com.powsybl.openloadflow.network.LfBranch;
+import com.powsybl.openloadflow.network.LfBus;
+import com.powsybl.openloadflow.network.LfNetwork;
+import com.powsybl.openloadflow.network.PiModel;
+import com.powsybl.openloadflow.network.VoltageControl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Didier Vidal {@literal <didier.vidal-ext at rte-france.com>}
+ */
+public class TransformerRatioManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransformerRatioManager.class);
+
+    private final boolean useInitialTapPosition;
+
+    public record ParallelRatioInfo(double rMax, double rMin, double rIni) {
```

Comment:
Parallel is confusing because reading your code, you only manage Transformers of the same transformer voltage control (merged elements).